class_ex/ex2/ex2.py

```diff
@@ -1,26 +1,50 @@
 class Person:
-    def __init__(self, name: str):
+    def __init__(self, name: str, age: int):
+        if not isinstance(name, str):
+            raise SyntaxError("Name has to be str type")
         self._name = name
+        self._age = age
 
+    @property
+    def name(self):
+        return self._name
 
-class Count(Person):
-    def __init__(self, name: str, cant: float = 0.0):
-        super().__init__(name)
-        if type(name) is not str:
-            raise SyntaxError("Name has to be str type")
+    @property
+    def age(self):
+        return self._age
+
+
+class Count:
+    def __init__(self, p1: Person, cant: float = 0.0):
+        if not isinstance(p1, Person):
+            raise SyntaxError("Name has to be a Person object")
+        self.p1 = p1
```

Comment:
```suggestion
        self.__p1 = p1
```
El objeto p1 debería ser privado y recuerda las privadas no es un simple _ sino doble _ '__'